apps/system/test/unit/activities_test.js

```diff
@@ -198,33 +238,196 @@ suite('system/Activities', function() {
+          id: 'foo',
+          type: 'activity-choice',
+          value: 0,
+          setAsDefault: false
+        };
+
+        sendEvent = this.sinon.stub(subject, '_sendEvent');
+        defaultAct = this.sinon.stub(DefaultActivityHelper, 'setDefaultAction');
+      });
+
+      test('without default activity set >', function() {
+        var set_default = false;
+        formatted.setAsDefault = set_default;
+        subject.choose(0, set_default);
+
+        assert.ok(sendEvent.calledWith(formatted),
+          'calls _sendEvent WITHOUT default activity set');
+        assert.isFalse(defaultAct.called, 'should not call the helper');
+      });
+
+      test('with a default activity set >', function() {
+        var set_default = true;
+        formatted.setAsDefault = set_default;
+        subject.choose(0, set_default);
+
+        assert.ok(sendEvent.calledWith(formatted),
+          'calls _sendEvent WITH default activity set');
+        assert.ok(defaultAct.calledWith('testactivity', 'testtype', 'manifest'),
+          'should call the helper with the proper values');
+      });
+    });
+
+    suite('cancel selection', function() {
+      test('calls _sendEvent', function() {
+        subject._detail = {
+          id: 'foo'
+        };
+        var stub = this.sinon.stub(subject, '_sendEvent');
+        var formatted = {
+          id: 'foo',
+          type: 'activity-choice',
+          value: -1
+        };
+        subject.cancel();
+        assert.ok(stub.calledWith(formatted));
+      });
     });
   });
 
-  suite('cancel', function() {
-    test('calls _sendEvent', function() {
-      subject._id = 'foo';
-      var stub = this.sinon.stub(subject, '_sendEvent');
-      var formatted = {
-        id: 'foo',
-        type: 'activity-choice',
-        value: -1
+  suite('when a new app is installed', function() {
+    var app,
+        listedName,
+        listedType;
+
+    setup(function() {
+      this.sinon.stub(DefaultActivityHelper, 'getDefaultAction');
+      this.sinon.stub(DefaultActivityHelper, 'setDefaultAction');
+
+      app = {
+        manifest: {
+          'activities': {
+            'browse': {
+              'filters': {
+                'type': 'photos'
+               }
+            },
+            'pick': {
+              'filters': {
+                'type': ['image/*']
+               }
+            },
+            'open': {
+              'filters': {
+                'type': ['video/*']
+               }
+            }
+          }
+        }
       };
-      subject.cancel();
-      assert.ok(stub.calledWith(formatted));
+
+      listedName = 'pick';
+      listedType = 'image/*';
+    });
+
+    test('manages the default launch for the app\'s activities', function() {
+      DefaultActivityHelper.getDefaultAction
+        .yields(null)
+        .withArgs(listedName, listedType).yields(
+          'app://fakeapp1.gaiamobile.org/manifest.webapp');
+
+      subject._onNewAppInstalled(app);
```

Comment:
instead of calling a private method here please dispatch an `applicationinstall` event, easier that faking the whole mozApp.mgmt too :)